src/MaxMind/Db/Reader.php

```diff
@@ -265,19 +265,18 @@ private function findMetadataStart($filename)
         $fileSize = $fstat['size'];
         $marker = self::$METADATA_START_MARKER;
         $markerLength = self::$METADATA_START_MARKER_LENGTH;
-        $metadataMaxLengthExcludingMarker
-            = min(self::$METADATA_MAX_SIZE, $fileSize) - $markerLength;
-
-        for ($i = 0; $i <= $metadataMaxLengthExcludingMarker; ++$i) {
-            for ($j = 0; $j < $markerLength; ++$j) {
-                fseek($handle, $fileSize - $i - $j - 1);
-                $matchBit = fgetc($handle);
-                if ($matchBit !== $marker[$markerLength - $j - 1]) {
-                    continue 2;
-                }
+
+        $minStart = $fileSize - min(self::$METADATA_MAX_SIZE + 1, $fileSize);
```

Comment:
Why +1?

Reply:
Because the offsets are 0 indexed so the last offset would be `$fileSize - 1`.

Reply:
Sorry, you are right. I'll change it tomorrow.